### PR TITLE
Feat: Add The Replica Dataset to DagsHub

### PR DESCRIPTION
**Pull request by samuelkb**
This PR contains the Readme file of the dataset with the following details.

The Replica Dataset is a dataset of high-quality reconstructions of a variety of indoor spaces. Each reconstruction has clean, dense geometry, high resolution, and high dynamic range textures, glass and mirror surface information, planar segmentation, as well as semantic class and instance segmentation.

 Paper with Code Page - https://paperswithcode.com/paper/the-replica-dataset-a-digital-replica-of
Dataset Homepage - https://arxiv.org/abs/1906.05797

Fixes https://github.com/DagsHub/3D-model-datasets/issues/10

### CODE REVIEW

File: TheReplicaDataset/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# The Replica Dataset
+
```
Comment (deanp70) [inline TheReplicaDataset/README.md:2]: Missing a link to the DagsHub Dataset project

**Review by deanp70 — CHANGES_REQUESTED** [review]
Hey @samuelkb, thanks for the awesome contribution. This dataset is really cool!

Can you please add a link to the DagsHub repo at the top of the readme, so people can easily find it. Then we can approve it!

File: TheReplicaDataset/README.md (hunk 1 above)
Comment (samuelkb) [reply TheReplicaDataset/README.md:3]: Link added in the last commit @deanp70

**Review by deanp70 — APPROVED** [review]
Awesome, approved and merging! Congrats on the great contribution

**Pull request merged**